:window: :wrench: Add api handlers for free connectors program

ambirdsall opened the pull request:
## What
Code to consume the new/updated APIs for the free connector program. Closes https://github.com/airbytehq/airbyte-cloud/issues/4006.

## How
The code in `.../cloud/lib/domain/freeConnectorProgram/api.ts` was generated by pointing `orval.config.ts` at the airbyte-cloud openAPI spec file, then deleting all unrelated code. `services/freeConnectorProgram.ts` wraps it with the necessary config for easier use.

## Recommended reading order
- `cloud/lib/domain/freeConnectorProgram/api.ts`
- `cloud/services/freeConnectorProgram.ts`

tealjulia commented on airbyte-webapp/src/packages/cloud/lib/domain/freeConnectorProgram/api.ts:
```diff
@@ -0,0 +1,40 @@
+  workspaceId: WorkspaceId;
+}
+
+/**
+ * Workspace Id from OSS Airbyte instance
+ */
```
OSS Airbyte Instance? (I know this one is auto-generated, but that seems mislabelled, right?)

ambirdsall replied:
Great catch!

josephkmh commented on airbyte-webapp/src/packages/cloud/services/freeConnectorProgram.ts:
```diff
@@ -0,0 +1,19 @@
+import { useConfig } from "./config";
+
+export const useFreeConnectorProgramInfo = () => {
+  const { workspaceId } = useCurrentWorkspace();
```
Usually I use `useCurrentWorkspaceId()` to just get the workspace ID. It's synchronous and just returns the id from the URL. `useCurrentWorkspace()` calls a suspense query, so it could create a waterfall request dependency in the worst case.

josephkmh reviewed (approved):
LGTM! Just one suggestion about using `useCurrentWorkspaceId()`

ambirdsall commented:
Pushed up some updates:
1) updated the code to use `useCurrentWorkspaceId`
2) moved the new files to within the `cloud/components/experiments/FreeConnectorProgram/` directory
3) Encapsulated the `showEnrollmentUi` logic within the `useFreeConnectorProgramInfo` hook
4) Added a stubbed/hardcoded `needsEmailVerification` value to unblock building out conditional UI for that use case (cf. the discussion in [this slack thread](https://airbytehq-team.slack.com/archives/C049XDAS19C/p1673875448547669))
